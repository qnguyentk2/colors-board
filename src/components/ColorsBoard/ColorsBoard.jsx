/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable no-debugger */
/* eslint-disable no-param-reassign */
import React, { Component } from 'react';

import styles from './ColorsBoard.scss';

const mainHues = [0, 30, 61, 89, 117, 150, 180, 210];

export const getRandomColor = (hue, lightness) => {
  const color = `hsl(${hue}, 100%, ${lightness}%)`;
  return color;
}

export const swapArrayElements = (arr, indexA, indexB) => {
  const temp = arr[indexA];
  arr[indexA] = arr[indexB];
  arr[indexB] = temp;
  return arr;
};

export const swapItemArrayInObject = (obj = {}, pointA = {}, pointB = {}, key) => {
  const itemA = `${key}${pointA.orderRow}`;
  const itemB = `${key}${pointB.orderRow}`;
  const temp = obj[itemA][pointA.orderCol];
  obj[itemA][pointA.orderCol] = obj[itemB][pointB.orderCol];
  obj[itemB][pointB.orderCol] = temp;

  return obj;
}

const randomColorRow = (lightness) => {
  return [...Array(8)].map((_, i) => ({
    color: getRandomColor(mainHues[i], lightness),
    order: i,
    board: 'shuffled'
  }))
}

class ColorsBoard extends Component {
  constructor(props) {
    super(props);

    const pieces2 = {
      item0: randomColorRow(90),
      item1: randomColorRow(80),
      item2: randomColorRow(70),
      item3: randomColorRow(60),
      item4: randomColorRow(50),
      item5: randomColorRow(40),
      item6: randomColorRow(30),
      item7: randomColorRow(20)
    }

    this.state = {
      pieces2,
      dragOrder: {},
      dropOrder: {},
    };
  }

  handleDragOver = (e) => {
    e.preventDefault();
  }

  handleDragStart = (e, idxCol, idxRow) => {
    const { dragOrder } = this.state;
    this.setState({
      dragOrder: {
        ...dragOrder,
        orderCol: idxCol,
        orderRow: idxRow,
      }
    })
  }

  handleDrop = (e, idxCol, idxRow) => {
    const { dropOrder } = this.state;
    this.setState({
      dropOrder: {
        ...dropOrder,
        orderCol: idxCol,
        orderRow: idxRow,
      }
    }, () => {
      const { dragOrder, pieces2 } = this.state;
      if (idxRow === dragOrder.orderRow) { 
        const newRowArray = swapArrayElements(pieces2[`item${dragOrder.orderRow}`], dragOrder.orderCol, idxCol);
        // reset state
        this.setState({
          pieces2: {
            ...pieces2,
            [`item${dragOrder.orderRow}`]: [...newRowArray]
          },
          dragOrder: {},
          dropOrder: {},
        })
      } else {
        const newPieces2 = swapItemArrayInObject(pieces2, dragOrder, this.state.dropOrder, 'item');
        this.setState({
          pieces2: {
            ...pieces2,
            ...newPieces2,
          },
          dragOrder: {},
          dropOrder: {},
        })
      }
    })
  }

  render() {
    const { pieces2 } = this.state;

    return (
      <div className={styles.dragContainer}>
        {
          Object.keys(pieces2).map((item, index) => {
            return pieces2[item].map((i, idx) => (
              <div
                // id={`drag_${index}`}
                key={`${item}-${i.color}`}
                draggable
                onDragStart={(e) => this.handleDragStart(e, idx, index)}
                onDragOver={this.handleDragOver}
                onDrop={(e) => this.handleDrop(e, idx, index)}
                className={styles.itemboxColor} 
                style={{ backgroundColor: `${i.color}`}}
              >
                {/* {index}, {idx} */}
              </div>
            ));
          })
        }
      </div>
    )
  }
}

export default ColorsBoard;
