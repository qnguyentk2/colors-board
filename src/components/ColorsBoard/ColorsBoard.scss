@import '../../../assets/styles/responsive.scss';

.dragContainer {
  text-align: center;
  display: inline-flex;
  width: 80%;
  justify-content: center;
  flex-wrap: wrap;
}
// .lineboxColor {

// }
.itemboxColor {
  // color: blue;
  // padding: 37px;
  width: 50px;
  height: 50px;

  @include Responsive(XS) {
    width: 30px;
    height: 30px;
	}
}
