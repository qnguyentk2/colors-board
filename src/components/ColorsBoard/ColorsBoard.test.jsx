/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import 'babel-polyfill';
import toJSON from 'enzyme-to-json';
import ColorsBoard, { 
  getRandomColor,
  swapArrayElements
} from './ColorsBoard';


const setup = () => {
  const wrapper = shallow(
    <ColorsBoard />
  );

  return { wrapper };
}

describe('Test getRandomColor', () => {
  test('it should return right color', () => {
    const result = 'hsl(aaa, 100%, 10%)'
    const colors = getRandomColor('aaa', 10);
    expect(colors).toEqual(result);
  })
})

describe('Test swapArrayElements', () => {
  test('it should return right arr', () => {
    const array = ['a','b'];
    const resultArray = ['b','a'];
    const result = swapArrayElements(array, 0, 1);
    expect(result).toEqual(resultArray);
  })
})

describe('Test AppContent', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('it should call handleDragOver', () => {
    const { wrapper } = setup();
    const e = {
      preventDefault: jest.fn().mockImplementation()
    }
    wrapper.instance().handleDragOver(e);
    expect(e.preventDefault).toHaveBeenCalled();
  });

  test('it should call handleDragStart', () => {
    const { wrapper } = setup();
    const e = {
      preventDefault: jest.fn
    }
    const mockSetState = jest.spyOn(wrapper, "setState")
    const dragStartFn = wrapper.instance().handleDragStart;
    dragStartFn(e, 0, 1);
    expect(mockSetState).toHaveBeenCalled();
  });

  test('it should call setState 1 time', () => {
    const { wrapper } = setup();
    const e = {
      preventDefault: jest.fn
    }
    const mockSetState = jest.spyOn(wrapper, "setState")
    wrapper.instance().handleDragStart(e, 0, 1);
    wrapper.instance().handleDrop(e, 0, 2);
    expect(mockSetState).toHaveBeenCalledTimes(3);
  });

  test('it should call swapArrayElements', () => {
    const { wrapper } = setup();
    const e = {
      preventDefault: jest.fn
    }
    const mockSetState = jest.spyOn(wrapper, "setState")
    wrapper.instance().handleDragStart(e, 0, 1);
    wrapper.instance().handleDrop(e, 0, 1);
    expect(mockSetState).toHaveBeenCalledTimes(3);
  });
})