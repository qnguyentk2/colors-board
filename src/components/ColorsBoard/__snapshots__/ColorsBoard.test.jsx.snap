// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AppContent renders without crashing 1`] = `
<div
  className="dragContainer"
>
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_0"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(0, 100%, 90%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_0"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(30, 100%, 90%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_0"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(61, 100%, 90%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_0"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(89, 100%, 90%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_0"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(117, 100%, 90%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_0"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(150, 100%, 90%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_0"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(180, 100%, 90%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_0"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(210, 100%, 90%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_1"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(0, 100%, 80%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_1"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(30, 100%, 80%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_1"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(61, 100%, 80%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_1"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(89, 100%, 80%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_1"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(117, 100%, 80%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_1"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(150, 100%, 80%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_1"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(180, 100%, 80%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_1"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(210, 100%, 80%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_2"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(0, 100%, 70%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_2"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(30, 100%, 70%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_2"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(61, 100%, 70%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_2"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(89, 100%, 70%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_2"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(117, 100%, 70%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_2"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(150, 100%, 70%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_2"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(180, 100%, 70%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_2"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(210, 100%, 70%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_3"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(0, 100%, 60%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_3"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(30, 100%, 60%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_3"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(61, 100%, 60%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_3"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(89, 100%, 60%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_3"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(117, 100%, 60%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_3"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(150, 100%, 60%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_3"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(180, 100%, 60%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_3"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(210, 100%, 60%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_4"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(0, 100%, 50%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_4"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(30, 100%, 50%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_4"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(61, 100%, 50%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_4"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(89, 100%, 50%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_4"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(117, 100%, 50%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_4"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(150, 100%, 50%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_4"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(180, 100%, 50%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_4"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(210, 100%, 50%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_5"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(0, 100%, 40%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_5"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(30, 100%, 40%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_5"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(61, 100%, 40%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_5"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(89, 100%, 40%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_5"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(117, 100%, 40%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_5"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(150, 100%, 40%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_5"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(180, 100%, 40%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_5"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(210, 100%, 40%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_6"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(0, 100%, 30%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_6"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(30, 100%, 30%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_6"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(61, 100%, 30%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_6"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(89, 100%, 30%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_6"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(117, 100%, 30%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_6"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(150, 100%, 30%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_6"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(180, 100%, 30%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_6"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(210, 100%, 30%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_7"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(0, 100%, 20%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_7"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(30, 100%, 20%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_7"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(61, 100%, 20%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_7"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(89, 100%, 20%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_7"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(117, 100%, 20%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_7"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(150, 100%, 20%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_7"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(180, 100%, 20%)",
      }
    }
  />
  <div
    className="itemboxColor"
    draggable={true}
    id="drag_7"
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "backgroundColor": "hsl(210, 100%, 20%)",
      }
    }
  />
</div>
`;
